### PR TITLE
Several improvements

### PR DESCRIPTION
**Pull request by alexmogavero**
I fixes few bugs especially regarding ESP32 support.
I also added an available topic to signal availability of node, based on last will MQTT feature.

I merged also the work done by @simone1999 in his fork.

I think this is a very useful library, it is worth to improve it over time.
Please let me know if there is anything else that can be done.

### CODE REVIEW

**Comment by PhracturedBlue** [comment]
This is a massive overhaul of the code.  It will take me quite a while to review it all.  Give me some time.

File: library.json [hunk 1]
```diff
@@ -7,37 +7,24 @@
   "platforms": ["espressif8266", "espressif32"],
   "authors":
   {
-    "name": "PhracturedBlue",
-    "url": "https://github.com/PhracturedBlue"
+    "name": "simone1999/PhracturedBlue",
```
Comment (PhracturedBlue) [inline library.json:10]: Please revert these changes

File: library.json [hunk 2]
```diff
@@ -7,37 +7,24 @@
...
+      "version": "1.1.3",
       "platforms": "espressif8266"
     },
     {
       "name": "AsyncMqttClient",
-      "version": "^0.8.1",
+      "version": "0.8.1",
       "platforms": "espressif8266"
-    },
-    {
```
Comment (PhracturedBlue) [inline library.json:29]: Is this no longer needed for ESP32 support?

**Review by PhracturedBlue — COMMENTED** [review]
I don't really see anything concerning in here.  Please fix the library.json, and it should be good to go.

File: library.json (hunk 1 above)
Comment (alexmogavero) [reply library.json]: I have reverted those changes

File: library.json (hunk 2 above)
Comment (alexmogavero) [reply library.json]: I think they are needed, I reverted to previous version. Thanks.

**Review by PhracturedBlue — APPROVED** [review]

**Comment by alexmogavero** [comment]
Thank you, I am glad to collaborate.
